projects/firmware_homework102/src/main.c

```diff
@@ -0,0 +1,60 @@
+#include <stdint.h>
+#include <stdlib.h>
+#include <time.h>
+
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+struct Counters {
+  uint8_t counter_a;
+  uint8_t counter_b;
+};
+
+struct Counters *counter_init() {
+  struct Counters *out = malloc(sizeof(struct Counters));
+  out->counter_a = 0;
+  out->counter_b = 0;
+  return out;
+}
+
+void delay(unsigned milliseconds) {
+  clock_t pause;
+  clock_t start;
+
+  pause = milliseconds * (CLOCKS_PER_SEC / 1000);
+  start = clock();
+  while ((clock() - start) < pause) {
+    continue;
+  }
+}
```

Comment:
This is quite creative, I'm glad you've got some C standard library knowledge. We do already have a few delay functions in our library (implemented using soft timers) though: `delay_us`, `delay_ms`, and `delay_s` from delay.h (in libraries/ms-common/inc).

A couple other points:
* we're using C11, you don't have to declare all variables at the start of the function (if that's what you were doing)
* please use types from <stdint.h> like `uint8_t`, `uint16_t` etc, normal types like `unsigned int` might have different sizes on x86 and stm32.